File: palindromic-substrings.cpp
#include <string>

using namespace std;

class Solution {
public:
	int countSubstrings(string s) {
		int count = 0;

		for (int i=0; i<s.size(); ++i) {
			int lt = i;
			int	rt = i;
			while ((0 <= lt) && (rt < s.size()) && (s[lt] == s[rt])) {
				++count;
				--lt;
				++rt;
			}
			
			lt = i;
			rt = i+1;
			while ((0 <= lt) && (rt < s.size()) && (s[lt] == s[rt])) {
				++count;
				--lt;
				++rt;
			}
		}

		return count;
    }
};

File: palindromic-substrings.py
class Solution:
    def countSubstrings(self, s: str) -> int:
        count = 0

        for i in range(len(s)):

            lt, rt = i, i
            while 0 <= lt and rt < len(s) and s[lt] == s[rt]:
                count += 1
                rt += 1
                lt -= 1
                
            lt, rt = i, i+1
            while 0 <= lt and rt < len(s) and s[lt] == s[rt]:
                count += 1
                rt += 1
                lt -= 1

        return count
